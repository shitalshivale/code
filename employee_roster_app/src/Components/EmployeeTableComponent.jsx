import React from "react";

const EmployeeTableComponent = ({ employees, currentPage, employeesPerPage, openModal }) => {

  const startIndex = (currentPage - 1) * employeesPerPage;
  const endIndex = startIndex + employeesPerPage;

  const paginatedEmployees = employees && employees.slice(startIndex, endIndex);
  
  if (!paginatedEmployees) {
    return null; 
  }


  return (
    <table className="rosterTable">
      <thead>
        <tr>
          <th>ID</th>
          <th>Name</th>
          <th>Contact No</th>
          <th>Address</th>
        </tr>
      </thead>
      <tbody>
        {paginatedEmployees && paginatedEmployees.map((employee) => {
            return(
                <tr key={employee.id} onClick={() => openModal(employee)}>
                <td>{employee.id}</td>
                <td>
                  <img
                    src={"/" + employee.empPhoto}
                    alt=''
                    height="30"
                    width="30"
                  />
                  {employee.empName}
                </td>
                <td>{employee.contactNo}</td>
                <td>{employee.address}</td>
              </tr>
            );
        })}
      </tbody>
    </table>
  );
};

export default EmployeeTableComponent;