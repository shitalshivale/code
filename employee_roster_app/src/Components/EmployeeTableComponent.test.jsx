// HelloWorld.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import EmployeeTableComponent from './EmployeeTableComponent';

const employees=[
  {
    "id": 1,
    "empName": "Deepti",
    "empLastName": "Sharma", 
    "contactNo": "9748117933",
    "address": "56 Sec 27",
    "empAge": 49,
    "jobTitle": "Analyst",
    "dateOfJoining": "2023-10-23",
    "jobDescription": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    "empPhoto": "./profile_dummy_img.jpg"
  },
  {
    "id": 2,
    "empName": "Kanika",
    "empLastName": "Singh", 
    "contactNo": "6260859447",
    "address": "10 Sec 2",
    "empAge": 58,
    "jobTitle": "Tester",
    "dateOfJoining": "2023-08-24",
    "jobDescription": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    "empPhoto": "./profile_dummy_img.jpg"
  },
]
const currentPage =1;
const employeesPerPage = 5;
const openModal= jest.fn();

describe('Employee table Component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(
        <EmployeeTableComponent 
            employees={employees} 
            currentPage={currentPage} 
            employeesPerPage={employeesPerPage}
            openModal={openModal}
        />
    );

    // Create a snapshot
    expect(asFragment()).toMatchSnapshot();
  });
  test('checks employee array length', () => {
    const { getAllByRole } = render(
        <EmployeeTableComponent 
            employees={employees} 
            currentPage={currentPage} 
            employeesPerPage={employeesPerPage}
            openModal={openModal}
        />
    );

    // Assuming each employee is rendered in a table row
    // eslint-disable-next-line testing-library/prefer-screen-queries
    const employeeRows = getAllByRole('row'); 
    expect(employeeRows.length).toBe(employees.length + 1); 
});
});