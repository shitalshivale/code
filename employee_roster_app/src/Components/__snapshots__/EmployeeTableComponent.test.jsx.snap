// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee table Component matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="rosterTable"
  >
    <thead>
      <tr>
        <th>
          ID â†‘
        </th>
        <th>
          Name 
        </th>
        <th>
          Contact No 
        </th>
        <th>
          Address 
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          <img
            alt=""
            height="30"
            src="./profile_dummy_img.jpg"
            width="30"
          />
          Deepti
        </td>
        <td>
          9748117933
        </td>
        <td>
          56 Sec 27
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          <img
            alt=""
            height="30"
            src="./profile_dummy_img.jpg"
            width="30"
          />
          Kanika
        </td>
        <td>
          6260859447
        </td>
        <td>
          10 Sec 2
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
